database/src/desktop/util_desktop_macos.mm

```diff
@@ -17,11 +17,47 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include "database/src/desktop/util_desktop.h"
+#include <string>
 
 namespace firebase {
 namespace database {
 namespace internal {
 
+// Split a string based on specified character delimiter into constituent parts
+static std::vector<std::string> split_string(const std::string& s,
+                                             const char delimiter='/') {
+  size_t pos = 0;
+  // This index is used as the starting index to search the delimiters from.
+  size_t delimiter_search_start = 0;
+  // Skip any leading delimiters
+  while(s[delimiter_search_start] == delimiter) {
```

Comment:
Can you use std::string.find here?

Reply:
I feel for this purpose, it will make the code real hard to follow given the edge cases.
From `find`, we can find the first position of delimiter but it could be a leading one or not. 
For the leading case, we will have to issue multiple `find`(s) repeatedly until we have found the last consecutive delimiter (Eg: ///Library/foo/bar)

Reply:
Gotcha.